File: src/main/java/net/Chidoziealways/everythingjapanese/recipe/ModRecipeBookCategories.java
package net.Chidoziealways.everythingjapanese.recipe;

import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.minecraft.core.Registry;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.item.crafting.RecipeBookCategory;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.RegistryObject;

public class ModRecipeBookCategories {
    public static final DeferredRegister<RecipeBookCategory> RECIPE_BOOK_CATEGORIES =
            DeferredRegister.create(Registries.RECIPE_BOOK_CATEGORY, EverythingJapanese.MOD_ID);

    public static final RegistryObject<RecipeBookCategory> GROWTH_CHAMBER = register("growth_chamber");

    private static RegistryObject<RecipeBookCategory> register(String pName) {
        return RECIPE_BOOK_CATEGORIES.register(pName, RecipeBookCategory::new);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/recipe/ModRecipeBookCategories.kt
package net.Chidoziealways.everythingjapanese.recipe

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.minecraft.core.registries.Registries
import net.minecraft.world.item.crafting.RecipeBookCategory
import net.neoforged.neoforge.registries.DeferredRegister
import thedarkcolour.kotlinforforge.neoforge.forge.getValue
import java.util.function.Supplier

object ModRecipeBookCategories {
    val RECIPE_BOOK_CATEGORIES = DeferredRegister.create(Registries.RECIPE_BOOK_CATEGORY, JAPANESE_MOD_ID)

    val GROWTH_CHAMBER by RECIPE_BOOK_CATEGORIES.register("growth_chamber", Supplier { RecipeBookCategory() })
}
